### PR TITLE
Dream 1717 ux for skills

### PR DESCRIPTION
**Pull request by Saga6569**

### CODE REVIEW

File: services/frontend/src/components/Menus/StatusToolTip/StatusToolTip.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import StatusToolTipAssistant from './StatusToolTipAssistant'
+import StatusToolTipCard from './StatusToolTipCard'
+
+const StatusToolTip = ({
+  name,
+  skill,
+  skills,
+  bot,
+}: {
+  name: string
```
Comment (artem-mar) [inline services/frontend/src/components/Menus/StatusToolTip/StatusToolTip.tsx:12]: name: 'assistantCard' | 'skillCard'
or
name: 'assistant' | 'skill'

File: services/frontend/src/utils/checkingAssistants.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    const message = `${i18n.t(
+      'error_message.prompt.lengthMax'
+    )} ${curentCountToken}/${maxToken}`
+    acc.error = [...acc.error, message]
+  }
+}
+
+export const examination = (data: ISkill) => {
+  const acc = {
+    error: [],
```
Comment (artem-mar) [inline services/frontend/src/utils/checkingAssistants.ts:208]:
```suggestion
    errors: [],
    warnings: [],
```

File: services/frontend/src/components/Menus/StatusToolTip/StatusToolTipCard.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import { ReactComponent as Error } from '@assets/icons/errorIcon.svg'
+import { ReactComponent as Warning } from '@assets/icons/warningIcon.svg'
+import { useTranslation } from 'react-i18next'
+import { Tooltip } from 'react-tooltip'
+import { ICollectionError } from 'types/types'
+import style from './StatusToolTip.module.scss'
+
+const StatusToolTipCard = ({
```
Comment (artem-mar) [inline services/frontend/src/components/Menus/StatusToolTip/StatusToolTipCard.tsx:8]: StatusTooltipSkill

File: services/frontend/src/components/Modules/AssistantModule/AssistantModule.tsx [hunk 4]
```diff
@@ -147,6 +150,14 @@ export const AssistantModule = () => {
...
   }, [bot])
 
+  const components = getAllComponents(bot?.name || '', {
+    refetchOnMount: true,
+  })
+
+  const resultExamination = examinationMessage(components)
+
+  const isPreviewTooltip = resultExamination.status === 'success' ? false : true
+
```
Comment (artem-mar) [inline services/frontend/src/components/Modules/AssistantModule/AssistantModule.tsx:160]:
```suggestion
  const isPreviewTooltip = resultExamination.status !== 'success'
```

File: services/frontend/src/components/Panels/AssistantDialogSidePanel/AssistantDialogSidePanel.tsx [hunk 5]
```diff
@@ -299,19 +309,25 @@ export const AssistantDialogSidePanel: FC<Props> = ({ dist }) => {
...
               props={{
                 onClick: handleDeploy,
-                disabled: deploy?.isLoading || deleteDeployment.isLoading,
+                disabled:
+                  deploy?.isLoading || deleteDeployment.isLoading || isError,
               }}
             >
               {t('sidepanels.assistant_dialog.btns.build_assistant')}
-            </Button>
+            </Button>{' '}
```
Comment (artem-mar) [inline services/frontend/src/components/Panels/AssistantDialogSidePanel/AssistantDialogSidePanel.tsx:330]:
```suggestion
            </Button>
```

File: services/frontend/src/components/Panels/BaseSidePanel/BaseSidePanel.tsx [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
-import SidePanel from 'components/Panels/SidePanel/SidePanel'
-import s from './BaseSidePanel.module.scss'
+import { ReactComponent as CloseIcon } from '@assets/icons/close.svg';
+import { useUIOptions } from 'context';
+import React, { FC, useEffect, useState } from 'react';
+import { useObserver } from 'hooks/useObserver';
+import { consts } from 'utils/consts';
+import SidePanel from 'components/Panels/SidePanel/SidePanel';
+import s from './BaseSidePanel.module.scss';
+import './BaseSidePanel.module.scss'
```
Comment (artem-mar) [inline services/frontend/src/components/Panels/BaseSidePanel/BaseSidePanel.tsx:8]: delete semicolons

File: services/frontend/src/components/Panels/WarningsInfo/WarningsInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+    return (
+      <div className={s.skill} key={i}>
+        <div className={s.skillBlock}>
+          <div
+            className={s.skillName}
+            onClick={e => handleEditBtnClick(e, skill.skill)}
+          >
+            <div className={`${s.messageCountError}`} style={colorError}>
+              {messageCountError}
```
Comment (artem-mar) [inline services/frontend/src/components/Panels/WarningsInfo/WarningsInfo.tsx:135]:
```suggestion
    const color = skill.data.error.length
      ? 'red'
      : skill.data.warning.length
      ? 'yellow'
      : ''

    const cx = classNames.bind(s)

    return (
      <div className={s.skill} key={i}>
        <div className={s.skillBlock}>
          <div
            className={s.skillName}
            onClick={e => handleEditBtnClick(e, skill.skill)}
          >
            <div className={cx(s.messageCountError, color)}>
```
and move styles to scss

File: services/frontend/src/components/Panels/WarningsInfo/WarningsInfo.module.scss [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+          background: white;
+          z-index: 1;
+
+          .messageCountError {
+            font-family: Inter;
+            font-size: 12px;
+            font-style: normal;
+            font-weight: 400;
+            line-height: 150%;
+          }
```
Comment (artem-mar) [inline services/frontend/src/components/Panels/WarningsInfo/WarningsInfo.module.scss:92]:
```suggestion
          .messageCountError {
            font-family: Inter;
            font-size: 12px;
            font-style: normal;
            font-weight: 400;
            line-height: 150%;

            &.red {
              color: #b20000;
            }
            &.yellow {
              color: #ff9500;
            }
          }
```

Comment (artem-mar) [reply services/frontend/src/components/Panels/WarningsInfo/WarningsInfo.module.scss:92]: the .warning and .error classes can be removed

File: services/frontend/src/utils/checkingAssistants.ts [hunk 9]
```diff
@@ -0,0 +1,254 @@
+import { franc } from 'https://esm.sh/franc@6'
```
Comment (artem-mar) [inline services/frontend/src/utils/checkingAssistants.ts:1]:
```suggestion
import { franc } from 'franc'
```

File: services/frontend/src/types/types.ts [hunk 10]
```diff
@@ -1,6 +1,28 @@
...
+export interface ICustomAssistant {
+  name: string
+  skill: ICustomSkill[]
+  bot: BotInfoInterface
+}
+export interface IСounter {
+  error: number
+  warning: number
+  [key: string]: number
+}
```
Comment (artem-mar) [inline services/frontend/src/types/types.ts:13]:
```suggestion
export interface IСounter {
  countError: number
  countWarning: number
  [key: string]: number
}
```

File: services/frontend/src/hooks/api/useComponent.ts [hunk 11]
```diff
@@ -124,6 +124,8 @@ export const useComponent = () => {
       skillDeleted()
       queryClient.invalidateQueries([ALL_COMPONENTS])
       queryClient.invalidateQueries([type, distName])
+      // queryClient.invalidateQueries(['skills_of_current_user_assistants']
```
Comment (artem-mar) [inline services/frontend/src/hooks/api/useComponent.ts:127]: ??

File: services/frontend/src/pages/Editor/EditorPage.tsx [hunk 12]
```diff
@@ -1,31 +1,17 @@
...
+import { RoutesList } from 'router/RoutesList';
+import { usePreview } from 'context/PreviewProvider';
+import { VISIBILITY_STATUS } from 'constants/constants';
+import { useAssistants } from 'hooks/api';
+import { consts } from 'utils/consts';
+import { AssistantModal, DeleteAssistantModal, DeleteSkillModal, PublishAssistantModal, ShareAssistantModal, SignInModal, SkillModal, SkillQuitModal, SkillsListModal } from 'components/Modals';
+import { BaseSidePanel } from 'components/Panels';
+import { Container } from 'components/UI';
+import { Sidebar } from 'components/Widgets';
+import { AssistantMenuInfo, DeepyHelperTab, IntegrationTab, SettingsTab, SkillsTab } from 'components/Widgets/Sidebar';
```
Comment (artem-mar) [inline services/frontend/src/pages/Editor/EditorPage.tsx:13]: delete semicolons

File: services/frontend/src/components/Widgets/Sidebar/AssistantMenuInfo.tsx [hunk 13]
```diff
@@ -0,0 +1,140 @@
...
+    }
+  }, [privateDistsInit.data])
+
+  let cx = classNames.bind(s)
+
+  const handleBtnClick = () => {
+    !copilotIsActive && deepyChatOpened()
+
+    trigger(TRIGGER_LEFT_SP_EVENT as any, {
+      children: <WarningsInfo />,
```
Comment (artem-mar) [inline services/frontend/src/components/Widgets/Sidebar/AssistantMenuInfo.tsx:48]:
```suggestion
    trigger(TRIGGER_LEFT_SP_EVENT, {
```

and extend the TEvents type with the value 'TRIGGER_LEFT_SP_EVENT''

File: services/frontend/src/components/Menus/StatusToolTip/StatusToolTipAssistant.tsx [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+  const count: IСounter = data.reduce(
+    (acc: IСounter, el: ICollectionError) => {
+      const error = el.error.length
+      const warning = el.warning.length
+      acc.error += error
+      acc.warning += warning
+      return acc
+    },
+    { error: 0, warning: 0 }
+  )
```
Comment (artem-mar) [inline services/frontend/src/components/Menus/StatusToolTip/StatusToolTipAssistant.tsx:31]:
```suggestion
  const count: IСounter = data.reduce(
    (acc: IСounter, el: ICollectionError) => {
      const error = el.error.length
      const warning = el.warning.length
      acc.countError += error
      acc.countWarning += warning
      return acc
    },
    { countError: 0, countWarning: 0 }
  )
```

**Review by artem-mar — APPROVED** [review]

**Review by NeoIsALie — APPROVED** [review]

**Pull request merged**